### PR TITLE
#686: Even less swifty JSON helper

### PR DESCRIPTION
**Pull request by bonkey**
How about now?

#686: Unified and simplified array and dictionary in JSON
#686: Simplified JSON: removed raw… values and JSONObjectifier
#686: Optimized JSON.isBool

### CODE REVIEW

**Comment by bonkey** [comment]
👍

**Comment by bonkey** [comment]
@hani-ibrahim 

Unfortunately, `testBillingAddressWithNoAddresses` and `testShippingAddressWithNoAddresses` are failing on my side as well.

And unfortunately, the tests are complex enough to not be easily debugged with just a quick look, so it would be great if you spot the reason.

**Comment by bonkey** [comment]
And Buddybuild passed because... it seems like it didn't run tests. There's

```
    Oops! There were test failures.
```

But no test logs. Argh!

**Comment by bonkey** [comment]
Maybe it's somehow related to the latest addition for [testing frameworks](https://www.buddybuild.com/blog/ios-framework-library-support)?

File: AtlasSDK/AtlasSDKTests/JSONNumbersTests.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import XCTest
+import Nimble
+
+@testable import AtlasSDK
+
+class JSONNumbersTests: JSONTestCase {
+
+    func testInt0() {
+        expect(0).to(equalJson(path: "numbers", "int0"))
```
Comment (bonkey) [inline AtlasSDK/AtlasSDKTests/JSONNumbersTests.swift]: @hani-ibrahim Compare it to your tests 😎

Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDKTests/JSONNumbersTests.swift]: 🙄

**Comment by hani-ibrahim** [comment]
@bonkey `testBillingAddressWithNoAddresses` is failing because as I said in my first comment, returning nil if the array is empty will be handled as invalid response so these **simple** two tests were failing because they were checking testBillingAddressWith**_NoAddresses_**

**Comment by hani-ibrahim** [comment]
But it's now much better ;)

File: AtlasSDK/AtlasSDK/AtlasAPIClient.swift [hunk 2]
```diff
@@ -39,8 +39,10 @@ public struct AtlasAPIClient {
 
     func fetch<Model: JSONInitializable>(from endpoint: Endpoint, completion: @escaping APIResultCompletion<[Model]>) {
         call(endpoint: endpoint, completion: completion) { response in
-            guard let json = response.body, let jsons = json.array.flatMap({ $0 }) else { return nil }
-            return jsons.flatMap { Model(json: $0) }
+            guard let json = response.body,
+                !json.array.isEmpty
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/AtlasAPIClient.swift]: I expected something like this. Assumed in hurry, that optional array + `flatMap()` == non-optional array + `!isEmpty`.

Fixed. And tests are passing.

**Comment by hani-ibrahim** [comment]
👍

**Comment by bonkey** [comment]
:+1 :

**Comment by bonkey** [comment]
Finally we can see all green...

**Pull request merged**